### PR TITLE
Social login: Kubernetes login code enhancements

### LINKED ISSUE #9871: A bad access_token with accessTokenSupported results in a 401 instead of a login panel
I have a valid oauth social login config with the accessTokenSupported attribute set to true (accessTokenRequired is NOT set).  I pass a simple string as the access_token on the request to access helloworld.  I was expecting to get the login screen, instead, I get a 401 response and and error message CWWKS5489E in the browser.  The log has: 
```
[11/19/19, 15:07:26:742 CST] 0000022b com.ibm.ws.security.social.tai.TAIUserApiUtils               W CWWKS5461E: The social login feature encountered an error while getting user information from the user API [https://localhost:8947/oidc/endpoint/OidcConfigSample/userinfo] that is configured for social login configuration [oidcLogin_accessTokenSupportedTrue_optionalParmsIncluded]. CWWKS5491E: Cannot process the error response from the [https://localhost:8947/oidc/endpoint/OidcConfigSample/userinfo] user API endpoint. The error was: [Parsing error: org.jose4j.json.internal.json_simple.parser.ParseException: Unexpected token END OF FILE at position 0.]
[11/19/19, 15:07:26:742 CST] 0000022b com.ibm.ws.security.social.error.SocialLoginException        E CWWKS5452E: The social login feature cannot authenticate the user because the response from the user API that is configured for the social login configuration [oidcLogin_accessTokenSupportedTrue_optionalParmsIncluded] is null or empty.
[11/19/19, 15:07:26:777 CST] 0000022b com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "com.ibm.ws.security.social.error.SocialLoginException: CWWKS5452E: The social login feature cannot authenticate the user because the response from the user API that is configured for the social login configuration [oidcLogin_accessTokenSupportedTrue_optionalParmsIncluded] is null or empty. com.ibm.ws.security.social.tai.OAuthLoginFlow 98" at ffdc_19.11.19_15.07.26.0.log
```

### PR DESCRIPTION
**Pull request by ayoho**
- Should fix #9871
- Adds new NLS messages
- Consolidates some duplicated code
- Enhances an existing error message when an HTTP response is bad

### CODE REVIEW

**Comment by ayoho** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.social_fat.delegated,com.ibm.ws.security.social_fat.multiProvider,com.ibm.ws.security.social_fat.oidcCertification

File: dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops [hunk 1]
```diff
@@ -481,8 +491,8 @@ RESPONSE_STATUS_MISSING_OR_ERROR=CWWKS5477E: The response status cannot be found
 RESPONSE_STATUS_MISSING_OR_ERROR.explanation=The request likely failed because it was unauthorized or missing information such as a required request parameter or header. If the response status was not null, the response status code and reason is included in the message.
 RESPONSE_STATUS_MISSING_OR_ERROR.useraction=Check the response status code in the message, if present. Verify that all required parameters and headers were included in the request.
 
-# 0=URL to which the request was submitted, 1=Error message
-RESPONSE_STATUS_UNSUCCESSFUL=CWWKS5478E: The request to the endpoint [{0}] failed. The error was: {1}
+# 0=URL to which the request was submitted, 1=HTTP response status (e.g. 404 Not Found), 2=Error message
+RESPONSE_STATUS_UNSUCCESSFUL=CWWKS5478E: The request to the endpoint [{0}] failed. The response status was [{1}]. The error was: {2}
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops]: Change
"The error was: {2}"
To
"The error is: {2}"

File: dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops [hunk 2]
```diff
@@ -66,23 +66,33 @@ ACCESS_TOKEN_MISSING_FROM_HEADERS.explanation=The social login configuration tha
...
+# 0=JSON key string, 1=JSON value type, for example, object, string, number, and so on, 2=JSON value type, 3=User API response string
+KUBERNETES_USER_API_RESPONSE_WRONG_JSON_TYPE=CWWKS5379E: The value for the key [{0}] in the Kubernetes user API response is expected to be of type {1}, but the value is of type {2}. The full user API response is [{3}].
 KUBERNETES_USER_API_RESPONSE_WRONG_JSON_TYPE.explanation=The value for the specified key did not have the correct type. The response from the user API might be malformed, or the user API returned a response with an unexpected structure.
-KUBERNETES_USER_API_RESPONSE_WRONG_JSON_TYPE.useraction=Check the structure of the response from the user API. Check the response from the user API to see if an error occurred.
+KUBERNETES_USER_API_RESPONSE_WRONG_JSON_TYPE.useraction=Check the structure of the response from the user API. Check the response from the user API to see whether an error occurred.
+
+# 0=Error string from user API response
+KUBERNETES_USER_API_RESPONSE_ERROR=CWWKS5380E: The Kubernetes user API encountered an error while processing the access token. The error is [{0}]
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops]: Change to:
"The Kubernetes user API encountered an error while it was processing the access token. The error is [{0}]"

File: dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops [hunk 3]
```diff
@@ -66,23 +66,33 @@ ACCESS_TOKEN_MISSING_FROM_HEADERS.explanation=The social login configuration tha
...
+KUBERNETES_USER_API_RESPONSE_ERROR.explanation=The response from the Kubernetes user API contains an error. The access token might not be valid, or some other error occurred while processing the token.
+KUBERNETES_USER_API_RESPONSE_ERROR.useraction=Check the error that is displayed in the message. Ensure that the access token is valid and is not malformed.
+
+# 0=Social login configuration ID, 1=Value of the userNameAttribute, 2=userNameAttribute, 3=Alternative key that will be used instead
+KUBERNETES_USER_API_RESPONSE_DEFAULT_USER_ATTR_NOT_FOUND=CWWKS5381W: The social login configuration [{0}] specifies [{1}] as the value for the [{2}] attribute, but the user API response does not contain a [{1}] key. The [{3}] key, if present in the user API response, is used to determine the user name instead.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops]: Change
"The [{3}] key, if present in the user API response, is used to determine the user name instead."
To
" The [{3}] key, if present in the user API response, is used to determine the username instead."

File: dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops [hunk 4]
```diff
@@ -66,23 +66,33 @@ ACCESS_TOKEN_MISSING_FROM_HEADERS.explanation=The social login configuration tha
...
+KUBERNETES_USER_API_RESPONSE_DEFAULT_USER_ATTR_NOT_FOUND.explanation=The user API response does not contain a key that corresponds to the value of the configuration attribute that is specified in the message. The social login feature attempts to use a different key instead.
+KUBERNETES_USER_API_RESPONSE_DEFAULT_USER_ATTR_NOT_FOUND.useraction=Verify that the configuration attribute that is specified in the message is configured correctly. Ensure that the value of the configuration attribute corresponds to an entry in the user API response that should be used as the user name.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.social/resources/com/ibm/ws/security/social/resources/SocialMessages.nlsprops]: Change
"Ensure that the value of the configuration attribute corresponds to an entry in the user API response that should be used as the user name."
To
"Ensure that the value of the configuration attribute corresponds to the entry in the user API response that should be used as the username."

**Comment by ayoho** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.social_fat,com.ibm.ws.security.social_fat.LibertyOP,com.ibm.ws.security.social_fat.delegated,com.ibm.ws.security.social_fat.multiProvider,com.ibm.ws.security.social_fat.oidcCertification

**Comment by ayoho** [comment]
#run-libby-bot

**Pull request merged**